File: backend/MessagesService/src/domain/common/requests.ts

interface createMessageRequest {
    access_token: string;
    sender_id: string;
    receiver_id: string;
    message: string;
}

interface connectionRequest {
    access_token: string;
    sender_id: string;
    receiver_id: string;
}

interface getLast10MessagesRequest {
    access_token: string;
    sender_id: string;
    receiver_id: string;
    last_timestamp: Date | undefined;
}

interface getChatsIdsRequest {
    access_token: string;
    user_id: string;
}

export { createMessageRequest, getLast10MessagesRequest, connectionRequest, getChatsIdsRequest };

File: backend/MessangeService/domain/entities/requests.go
package entities

type Get10LastRequest struct {
	SenderId    string `json:"sender_id"`
	ReceiverId  string `json:"receiver_id"`
	Message     string `json:"message"`
	Date        string `json:"date"`
	AccessToken string `json:"access_token"`
}

type GetChatIdsRequest struct {
	SenderId    string `json:"sender_id"`
	AccessToken string `json:"access_token"`
}

type VerifycationRequest struct {
	AccessToken string `json:"access_token"`
}
